core/jazz_slack-channel/index.js

```diff
@@ -0,0 +1,478 @@
+	logger.init(event, context);
+	var channelMembers = [], isServAccRequested = true;
+
+	genericInputValidation(event, config)
+	.then(() => createPublicSlackChannel(config, event.body))
+	.then(res => addMemberToChannel (config, res.data, isServAccRequested, event.body))
+	.then(res => {
+		logger.info("successfully created public slack channel: " + JSON.stringify(res));
+		var result = {
+			"channel_id" : res.data.channelInfo.id,
+			"channel_name" : res.data.channelInfo.name,
+			"channel_link" : res.data.channelInfo.link,
+			"members" : res.data.slackChannelMembers,
+		};
+		return cb(null, responseObj(result, event.body));
+	})
+	.catch(error => {
+		logger.error(error);
+		if(error.result === "inputError") {
+			return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+		} else if(error.result === "unauthorized") {
+			return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+		} else {
+			return cb(JSON.stringify(errorHandler.throwInternalServerError(error.message)));
+		}
+	});
+};
+
+function genericInputValidation(event, config) {
+  logger.debug("Inside genericInputValidation");
+	return new Promise((resolve, reject) => {
+		if(!event || !event.body || !event.method){
+			reject({
+				result: "inputError",
+				message: "Service inputs not defined!"
+			});
+		}
+
+		if(!event.body.channel_name){
+			reject({
+				result: "inputError",
+				message: "channel_name is required!."
+			});
+		} else {
+			var name = event.body.channel_name;
+			if(name.length > config.slack_channel_name_length){
+				reject({
+					result : "inputError",
+					message : "channel name should not exceed more than 21 characters"
+				});
+			} else if( /[^a-zA-Z0-9\-\_]/.test(name) ){
+				reject({
+					result : "inputError",
+					message : "channel name should accept letters, numbers, hyphens, and underscores"
+				});
+			}
+		}
+
+		if( !event.body.users || event.body.users.length < 1){
+			reject({
+				result: "inputError",
+				message: "users.email_id is required!."
+			});
+		} else {
+			var users = event.body.users;
+			for (var i in users){
+				var emailAddress = users[i].email_id;
+				if(!validateEmailId(emailAddress)){
+					var msg = 'Not a valid email id!';
+					if(emailAddress){
+						msg = emailAddress +" not a valid email id!.";
+					}
+					reject({
+						result: "inputError",
+						message: msg
+					});
+				} else if(emailAddress.toLowerCase() === (config.service_account_emailId).toLowerCase()){
+          isServAccRequested = true;
+        }
+			}
+		}
+
+		if(event.method && event.method !== "POST") {
+			reject({
+				result: "inputError",
+				message: "Unsupported method/request."
+			});
+		}
+
+		if(!event.principalId) {
+			reject({
+				result: "unauthorized",
+				message: "Unauthorized."
+			});
+		}
+
+		resolve();
+	});
+}
+
+function identifyMembers (members, type, value){
+  var info = null;
+
+  members.forEach(each => {
```

Comment:
you are actually finding a member matching a criteria. Better way to do this would be something like this
var out = members.find(each => {
  var memberEmailId = (each.profile.email) ? each.profile.email : null;
  if(memberEmailId && type === 'id' && each.id === value ){
  return formatData(each);
} else if(memberEmailId && type === 'email' && memberEmailId.toLowerCase() === value.toLowerCase()){
  return formatData(each);
  }
})

what do you think?

Reply:
Done